src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Service/Gateway/LoginService.php

```diff
@@ -128,6 +128,9 @@ public function singleSignOn(Request $httpRequest, ReceivedAuthnRequest $origina
 
         // Check if the NameID is provided and we may use it.
         $nameId = $originalRequest->getNameId();
+        if (!$nameId) {
+            throw new RequesterFailureException('The NameId was not present in the AuthNRequest');
+        }
```

Comment:
This change seems unrelated to the story, and it is. 

During testing of the simulated ADFS authnrequests/samlresponse I ran into a situation where the nameId was not present in the AuthNRequest. And that resulted in an uncaught error. I added this `RequesterFailureException` to get a somewhat more detailed error report in the logs.

/boyscout rule

Reply:
:100: and thanks for the heads up!